lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -392,8 +392,12 @@ e2function number entity:getSkin()
 end
 
 --- Sets <this>'s skin number.
-e2function void entity:setSkin(skin)
-	if IsValid(this) then this:SetSkin(skin) end
+e2function void entity:setSkin(skinIndex)
+	if not IsValid(this) or this:IsPlayer() or not isOwner(self, this) or not gamemode.Call("CanProperty", self.player, "skin", this) then return end
+	if isentity(this.AttachedEntity) and IsValid(this.AttachedEntity) then this = this.AttachedEntity end -- If <this> has an attached entity, we want to modify its skin instead
```

Comment:
Slipping in this `AttachedEntity` logic in here makes this PR less trivial, and means that `getSkin` and `setSkin` might get/set the skins of different entities. Probably easiest to leave it out, and address it in another PR if it's actually needed.